common/src/main/java/org/apache/falcon/entity/EntityUtil.java

```diff
@@ -1128,22 +1129,26 @@ public static Date getPreviousInstanceTime(Date startTime, Frequency frequency,
     }
 
 
-    public static List<Date> getInstancesInBetween(Date startTime, Frequency frequency, TimeZone timeZone,
+    public static List<Date> getInstancesInBetween(Date startTime, Date endTime, Frequency frequency, TimeZone timeZone,
                                                   Date startRange, Date endRange) {
         List<Date> result = new LinkedList<>();
+        if (endRange.before(startRange)) {
+            return result;
+        }
         if (timeZone == null) {
             timeZone = TimeZone.getTimeZone("UTC");
         }
-
         Date current = getPreviousInstanceTime(startTime, frequency, timeZone, startRange);
         while (true) {
+            if (!current.before(startRange) && !current.after(endRange)
```

Comment:
both startRange and endRange can be out of the start time and end time in which case the instances from the start time till endtime(Excluding end time) need to be returned)